commit 43b190a8d20a5d64e710a20e053f9a5ffcca9e5d
Author: Lars Vogel <Lars.Vogel@vogella.com>
Date:   Tue Dec 13 11:32:03 2016 +0100

    Bug 509135 - Use varargs in ControlExample#getResourceString to match
    MessageFormat.format
    
    Change-Id: Ib86415f8bf4d496dd19fe51f17c3f34d8d3305e4
    Signed-off-by: Lars Vogel <Lars.Vogel@vogella.com>

2	2	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/BrowserTab.java
1	1	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ColorTab.java
1	1	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
1	1	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/DialogTab.java
6	6	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/MenuTab.java
1	1	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/Tab.java
1	1	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TabFolderTab.java
1	1	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TableTab.java
1	1	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/BrowserTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/BrowserTab.java
index 9a12879..68ee7ab 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/BrowserTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/BrowserTab.java
@@ -104,7 +104,7 @@ class BrowserTab extends Tab {
 			}
 			MessageBox dialog = new MessageBox(shell, SWT.ICON_WARNING | SWT.OK);
 			String resourceString = (style & SWT.MOZILLA) != 0 ? "MozillaNotFound" : "WebKitNotFound";
-			dialog.setMessage(ControlExample.getResourceString(resourceString, new String [] {e.getMessage()}));
+			dialog.setMessage(ControlExample.getResourceString(resourceString, e.getMessage()));
 			dialog.open();
 		}
 
@@ -209,7 +209,7 @@ class BrowserTab extends Tab {
 			public void widgetSelected(SelectionEvent e) {
 				if (errorMessage != null && tabFolder.getSelection()[0].getText().equals(getTabText())) {
 					MessageBox dialog = new MessageBox(shell, SWT.ICON_WARNING | SWT.OK);
-					dialog.setMessage(ControlExample.getResourceString("BrowserNotFound", new String [] {errorMessage}));
+					dialog.setMessage(ControlExample.getResourceString("BrowserNotFound", errorMessage));
 					dialog.open();
 				}
 			}
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ColorTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ColorTab.java
index fe6cdd0..4d5f8a0 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ColorTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ColorTab.java
@@ -108,7 +108,7 @@ class ColorTab extends Tab {
 		for (String columnTitle : columnTitles) {
 			TableColumn tableColumn = new TableColumn(colors, SWT.NONE);
 			tableColumn.setText(columnTitle);
-			tableColumn.setToolTipText(ControlExample.getResourceString("Tooltip", new String [] {columnTitle}));
+			tableColumn.setToolTipText(ControlExample.getResourceString("Tooltip", columnTitle));
 		}
 		// fill in the Data. Put an empty line inbetween "Named" and "SWT" colors.
 		boolean emptyLineFlag=false;
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
index 86b2c39..3e85e4c 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
@@ -170,7 +170,7 @@ public class ControlExample {
 	 * with the given arguments. If the key is not found,
 	 * return the key.
 	 */
-	static String getResourceString(String key, Object[] args) {
+	static String getResourceString(String key, Object... args) {
 		try {
 			return MessageFormat.format(getResourceString(key), args);
 		} catch (MissingResourceException e) {
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/DialogTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/DialogTab.java
index 40b564b..8f19139 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/DialogTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/DialogTab.java
@@ -151,7 +151,7 @@ class DialogTab extends Tab {
 			dialog.setText (ControlExample.getResourceString("Title"));
 			RGB result = dialog.open ();
 			textWidget.append (ControlExample.getResourceString("ColorDialog") + Text.DELIMITER);
-			textWidget.append (ControlExample.getResourceString("Result", new String [] {"" + result}) + Text.DELIMITER);
+			textWidget.append (ControlExample.getResourceString("Result", "" + result) + Text.DELIMITER);
 			textWidget.append ("getRGB() = " + dialog.getRGB() + Text.DELIMITER);
 			textWidget.append ("getRGBs() =" + Text.DELIMITER);
 			RGB[] rgbs = dialog.getRGBs();
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/MenuTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/MenuTab.java
index d42a6e2..2b4d63a 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/MenuTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/MenuTab.java
@@ -91,7 +91,7 @@ class MenuTab extends Tab {
 				MenuItem item = new MenuItem(menuBar, SWT.CASCADE);
 				item.setText(getMenuItemText("Cascade"));
 				if (imagesButton.getSelection()) item.setImage(instance.images[ControlExample.ciOpenFolder]);
-				if (tooltipButton.getSelection()) item.setToolTipText(ControlExample.getResourceString("Tooltip", new String[] {item.getText() }));
+				if (tooltipButton.getSelection()) item.setToolTipText(ControlExample.getResourceString("Tooltip", item.getText() ));
 				hookListeners(item);
 				Menu dropDownMenu = new Menu(shell, SWT.DROP_DOWN | radioBehavior);
 				item.setMenu(dropDownMenu);
@@ -254,13 +254,13 @@ class MenuTab extends Tab {
 			if (acceleratorsButton.getSelection()) item.setAccelerator(SWT.MOD1 + SWT.MOD2 + 'P');
 			if (imagesButton.getSelection()) item.setImage(instance.images[ControlExample.ciClosedFolder]);
 			item.setEnabled(enabledButton.getSelection());
-			if (tooltipButton.getSelection()) item.setToolTipText(ControlExample.getResourceString("Tooltip", new String[] {item.getText() }));
+			if (tooltipButton.getSelection()) item.setToolTipText(ControlExample.getResourceString("Tooltip", item.getText() ));
 			hookListeners(item);
 		}
 
 		if (separatorButton.getSelection()) {
 			item = new MenuItem(menu, SWT.SEPARATOR);
-			if (tooltipButton.getSelection()) item.setToolTipText(ControlExample.getResourceString("Tooltip", new String[] {item.getText() }));
+			if (tooltipButton.getSelection()) item.setToolTipText(ControlExample.getResourceString("Tooltip", item.getText() ));
 		}
 
 		if (checkButton.getSelection()) {
@@ -269,7 +269,7 @@ class MenuTab extends Tab {
 			if (acceleratorsButton.getSelection()) item.setAccelerator(SWT.MOD1 + SWT.MOD2 + 'C');
 			if (imagesButton.getSelection()) item.setImage(instance.images[ControlExample.ciOpenFolder]);
 			item.setEnabled(enabledButton.getSelection());
-			if (tooltipButton.getSelection()) item.setToolTipText(ControlExample.getResourceString("Tooltip", new String[] {item.getText() }));
+			if (tooltipButton.getSelection()) item.setToolTipText(ControlExample.getResourceString("Tooltip", item.getText() ));
 			hookListeners(item);
 		}
 
@@ -280,7 +280,7 @@ class MenuTab extends Tab {
 			if (imagesButton.getSelection()) item.setImage(instance.images[ControlExample.ciTarget]);
 			item.setSelection(true);
 			item.setEnabled(enabledButton.getSelection());
-			if (tooltipButton.getSelection()) item.setToolTipText(ControlExample.getResourceString("Tooltip", new String[] {item.getText() }));
+			if (tooltipButton.getSelection()) item.setToolTipText(ControlExample.getResourceString("Tooltip", item.getText() ));
 			hookListeners(item);
 
 			item = new MenuItem(menu, SWT.RADIO);
@@ -302,7 +302,7 @@ class MenuTab extends Tab {
 			item.setMenu(subMenu);
 			item.setEnabled(enabledButton.getSelection());
 			hookListeners(subMenu);
-			if (tooltipButton.getSelection()) item.setToolTipText(ControlExample.getResourceString("Tooltip", new String[] {item.getText() }));
+			if (tooltipButton.getSelection()) item.setToolTipText(ControlExample.getResourceString("Tooltip", item.getText() ));
 			createMenuItems(subMenu, createSubSubMenu, false);
 		}
 	}
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/Tab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/Tab.java
index 9e105b6..910988e 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/Tab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/Tab.java
@@ -618,7 +618,7 @@ abstract class Tab {
 		dialog.setLayout(new GridLayout());
 		dialog.setText(ControlExample.getResourceString ("Edit_Event"));
 		Label label = new Label (dialog, SWT.NONE);
-		label.setText (ControlExample.getResourceString ("Edit_Event_Fields", new String [] {EVENT_INFO[index].name}));
+		label.setText (ControlExample.getResourceString ("Edit_Event_Fields", EVENT_INFO[index].name));
 
 		Group group = new Group (dialog, SWT.NONE);
 		group.setLayout(new GridLayout(2, false));
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TabFolderTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TabFolderTab.java
index dff3654..df0f02f 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TabFolderTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TabFolderTab.java
@@ -72,7 +72,7 @@ class TabFolderTab extends Tab {
 		for (int i = 0; i < TabItems1.length; i++) {
 			TabItem item = new TabItem(tabFolder1, SWT.NONE);
 			item.setText(TabItems1[i]);
-			item.setToolTipText(ControlExample.getResourceString("Tooltip", new String [] {TabItems1[i]}));
+			item.setToolTipText(ControlExample.getResourceString("Tooltip", TabItems1[i]));
 			Text content = new Text(tabFolder1, SWT.WRAP | SWT.MULTI);
 			content.setText(ControlExample.getResourceString("TabItem_content") + ": " + i);
 			item.setControl(content);
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TableTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TableTab.java
index 6387fcf..2801d3d 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TableTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TableTab.java
@@ -321,7 +321,7 @@ class TableTab extends ScrollableTab {
 			for (int i = 0; i < columnTitles.length; i++) {
 				TableColumn tableColumn = new TableColumn(table1, SWT.NONE);
 				tableColumn.setText(columnTitles[i]);
-				tableColumn.setToolTipText(ControlExample.getResourceString("Tooltip", new String [] {columnTitles[i]}));
+				tableColumn.setToolTipText(ControlExample.getResourceString("Tooltip", columnTitles[i]));
 				if (headerImagesButton.getSelection()) tableColumn.setImage(instance.images [i % 3]);
 			}
 			table1.setSortColumn(table1.getColumn(0));
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java
index c80619d..965ca35 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java
@@ -325,7 +325,7 @@ class TreeTab extends ScrollableTab {
 			for (String columnTitle : columnTitles) {
 				TreeColumn treeColumn = new TreeColumn(tree1, SWT.NONE);
 				treeColumn.setText(columnTitle);
-				treeColumn.setToolTipText(ControlExample.getResourceString("Tooltip", new String [] {columnTitle}));
+				treeColumn.setToolTipText(ControlExample.getResourceString("Tooltip", columnTitle));
 			}
 			tree1.setSortColumn(tree1.getColumn(0));
 		}
commit 88c6ff9c59f4e0512cb3710f7bce1fd5d15a284b
Author: Alexander Kurtakov <akurtako@redhat.com>
Date:   Wed Dec 14 15:01:54 2016 +0200

    Bug 509135 - Use varargs in ControlExample#getResourceString to match
    MessageFormat.format
    
    Fix warnings in code using the method.
    
    Change-Id: Ib8ac199d325d90ee89d82fa9f7b3de8c7d921264
    Signed-off-by: Alexander Kurtakov <akurtako@redhat.com>

12	12	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/DialogTab.java
1	1	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/MenuTab.java
1	1	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/DialogTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/DialogTab.java
index 8f19139..aa7cbd7 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/DialogTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/DialogTab.java
@@ -175,7 +175,7 @@ class DialogTab extends Tab {
 			dialog.setText (ControlExample.getResourceString("Title"));
 			String result = dialog.open ();
 			textWidget.append (ControlExample.getResourceString("DirectoryDialog") + Text.DELIMITER);
-			textWidget.append (ControlExample.getResourceString("Result", new String [] {"" + result}) + Text.DELIMITER + Text.DELIMITER);
+			textWidget.append (ControlExample.getResourceString("Result", "" + result) + Text.DELIMITER + Text.DELIMITER);
 			directoryDialogResult = result;
 			return;
 		}
@@ -191,7 +191,7 @@ class DialogTab extends Tab {
 			dialog.setText (ControlExample.getResourceString("Title"));
 			String result = dialog.open();
 			textWidget.append (ControlExample.getResourceString("FileDialog") + Text.DELIMITER);
-			textWidget.append (ControlExample.getResourceString("Result", new String [] {"" + result}) + Text.DELIMITER);
+			textWidget.append (ControlExample.getResourceString("Result", "" + result) + Text.DELIMITER);
 			textWidget.append ("getFilterIndex() =" + dialog.getFilterIndex() + Text.DELIMITER);
 			textWidget.append ("getFilterPath() =" + dialog.getFilterPath() + Text.DELIMITER);
 			textWidget.append ("getFileName() =" + dialog.getFileName() + Text.DELIMITER);
@@ -216,7 +216,7 @@ class DialogTab extends Tab {
 			dialog.setText (ControlExample.getResourceString("Title"));
 			FontData result = dialog.open ();
 			textWidget.append (ControlExample.getResourceString("FontDialog") + Text.DELIMITER);
-			textWidget.append (ControlExample.getResourceString("Result", new String [] {"" + result}) + Text.DELIMITER);
+			textWidget.append (ControlExample.getResourceString("Result", "" + result) + Text.DELIMITER);
 			textWidget.append ("getFontList() =" + Text.DELIMITER);
 			FontData [] fonts = dialog.getFontList ();
 			if (fonts != null) {
@@ -239,7 +239,7 @@ class DialogTab extends Tab {
 			dialog.setText(ControlExample.getResourceString("Title"));
 			PrinterData result = dialog.open ();
 			textWidget.append (ControlExample.getResourceString("PrintDialog") + Text.DELIMITER);
-			textWidget.append (ControlExample.getResourceString("Result", new String [] {"" + result}) + Text.DELIMITER);
+			textWidget.append (ControlExample.getResourceString("Result", "" + result) + Text.DELIMITER);
 			if (result != null) {
 				textWidget.append ("printerData.scope = " + (result.scope == PrinterData.PAGE_RANGE ? "PAGE_RANGE" : result.scope == PrinterData.SELECTION ? "SELECTION" : "ALL_PAGES") + Text.DELIMITER);
 				textWidget.append ("printerData.startPage = " + result.startPage + Text.DELIMITER);
@@ -268,28 +268,28 @@ class DialogTab extends Tab {
 			 */
 			switch (result) {
 				case SWT.OK:
-					textWidget.append (ControlExample.getResourceString("Result", new String [] {"SWT.OK"}));
+					textWidget.append (ControlExample.getResourceString("Result", "SWT.OK"));
 					break;
 				case SWT.YES:
-					textWidget.append (ControlExample.getResourceString("Result", new String [] {"SWT.YES"}));
+					textWidget.append (ControlExample.getResourceString("Result", "SWT.YES"));
 					break;
 				case SWT.NO:
-					textWidget.append (ControlExample.getResourceString("Result", new String [] {"SWT.NO"}));
+					textWidget.append (ControlExample.getResourceString("Result", "SWT.NO"));
 					break;
 				case SWT.CANCEL:
-					textWidget.append (ControlExample.getResourceString("Result", new String [] {"SWT.CANCEL"}));
+					textWidget.append (ControlExample.getResourceString("Result", "SWT.CANCEL"));
 					break;
 				case SWT.ABORT:
-					textWidget.append (ControlExample.getResourceString("Result", new String [] {"SWT.ABORT"}));
+					textWidget.append (ControlExample.getResourceString("Result", "SWT.ABORT"));
 					break;
 				case SWT.RETRY:
-					textWidget.append (ControlExample.getResourceString("Result", new String [] {"SWT.RETRY"}));
+					textWidget.append (ControlExample.getResourceString("Result", "SWT.RETRY"));
 					break;
 				case SWT.IGNORE:
-					textWidget.append (ControlExample.getResourceString("Result", new String [] {"SWT.IGNORE"}));
+					textWidget.append (ControlExample.getResourceString("Result", "SWT.IGNORE"));
 					break;
 				default:
-					textWidget.append(ControlExample.getResourceString("Result", new String [] {"" + result}));
+					textWidget.append(ControlExample.getResourceString("Result", "" + result));
 					break;
 			}
 			textWidget.append (Text.DELIMITER + Text.DELIMITER);
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/MenuTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/MenuTab.java
index 2b4d63a..897d1cc 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/MenuTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/MenuTab.java
@@ -288,7 +288,7 @@ class MenuTab extends Tab {
 			if (acceleratorsButton.getSelection()) item.setAccelerator(SWT.MOD1 + SWT.MOD2 + '2');
 			if (imagesButton.getSelection()) item.setImage(instance.images[ControlExample.ciTarget]);
 			item.setEnabled(enabledButton.getSelection());
-			if (tooltipButton.getSelection()) item.setToolTipText(ControlExample.getResourceString("Tooltip", new String[] {item.getText() }));
+			if (tooltipButton.getSelection()) item.setToolTipText(ControlExample.getResourceString("Tooltip", item.getText()));
 			hookListeners(item);
 		}
 
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java
index 965ca35..6fd29a0 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java
@@ -356,7 +356,7 @@ class TreeTab extends ScrollableTab {
 			for (int i = 0; i < columnTitles.length; i++) {
 				TreeColumn treeColumn = new TreeColumn(tree2, SWT.NONE);
 				treeColumn.setText(columnTitles[i]);
-				treeColumn.setToolTipText(ControlExample.getResourceString("Tooltip", new String [] {columnTitles[i]}));
+				treeColumn.setToolTipText(ControlExample.getResourceString("Tooltip", columnTitles[i]));
 				if (headerImagesButton.getSelection()) treeColumn.setImage(instance.images [i % 3]);
 			}
 		}
